### PR TITLE
Adding B-physics validation

### PR DESCRIPTION
**Pull request by inugent**
Adding B-Physics Validation requested by Ye Chen.

### CODE REVIEW

**Comment by inugent** [comment]
Hello, 
What is the timeline for this to be integrated into CMSSW. Ye Chan requested this with as a high priority item and would like it also back ported to 7_1_X? Please advise?
Thank you and best regards,
Ian

**Comment by covarell** [comment]
+1

**Comment by deguio** [comment]
+1
ciao @inugent 
a couple of comments:
- is it necessary to add an entire analyzer to fill a single monitor element? referring to the spectrum one
- I don't particularly like the fact that the iBooker is stored in the DQMHelper class, but I guess it is ok provided it is not used outside the bookHistograms and this is the case

File: Validation/EventGenerator/plugins/BPhysicsSpectrum.cc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//
+///////////////////////////////////////
+ 
+#include "Validation/EventGenerator/interface/BPhysicsSpectrum.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"  
+
+#include "CLHEP/Units/defs.h"
+#include "CLHEP/Units/PhysicalConstants.h"
+
```
Comment (davidlange6) [inline Validation/EventGenerator/plugins/BPhysicsSpectrum.cc]: why are the CLHEP includes needed?

File: Validation/EventGenerator/interface/BPhysicsValidation.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  virtual void bookHistograms(DQMStore::IBooker &i, edm::Run const &, edm::EventSetup const &) override;
+  virtual void dqmBeginRun(const edm::Run& r, const edm::EventSetup& c) override;
+  virtual void analyze(edm::Event const&, edm::EventSetup const&) override;
+  
+ private:
+  
+  class ParticleMonitor{
+  public:
+  ParticleMonitor(TString _name,const edm::ParameterSet &_p):p(_p),name(_name),pdgid(p.getParameter<int>("pdgid")){};
```
Comment (davidlange6) [inline Validation/EventGenerator/interface/BPhysicsValidation.h]: no variables that start with _ please

File: Validation/EventGenerator/interface/DQMHelper.h [hunk 3]
```diff
@@ -17,6 +17,9 @@ class DQMHelper{
  public: 
   DQMHelper(DQMStore::IBooker *i);
   virtual ~DQMHelper();
+
+  MonitorElement* book1dHisto(TString name,TString title,int n,double xmin,double xmax,TString xaxis, TString yaxis);
+  MonitorElement* book2dHisto(TString name,TString title,int nx,double xmin,double xmax,int ny,double ymin,double ymax,TString xaxis, TString yaxis);
```
Comment (davidlange6) [inline Validation/EventGenerator/interface/DQMHelper.h]: whats the need for these functions? I see lots of string->TString->string conversions that might be better just avoided.

**Comment by inugent** [comment]
replaced by 8676

**Pull request closed without merging**